### PR TITLE
chore(card-group): fixed card border-top and grid mode transition

### PR DESCRIPTION
**Pull request by BrunnoM7**
### Related Ticket(s)

[Card group] Style bugs in Web components #6234

### Description

{{Add a human-readable description / detail summary of what the PR is changing and any details around how and why}}

{{If applicable, include a screenshot indicating an example or examples of what the PR is changing in the application}}

### Changelog

**New**

- {{new thing}}

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

**Comment by photodow** [comment]
- [ ] Seeing some adjustments to the footer?
image

- [ ] card section offset looks like it's missing a bottom border on the top card.

image

- [ ] border oddity when looking at card group with 6 of them or greater.

image

**Comment by photodow** [comment]
Looking good, thanks for working hard on this!

- [ ] other grid modes don't seem to be working (e.g. border, wide, condensed). I'm not sure if that was introduce here, in a previous PR, or if haven't gotten around to these yet.
- [ ] Outline cards don't appear to be looking correctly. Also images don't appear to be aligned in at least one spot.
image
- [ ] the borders looking good in 3 columns, but I'm seeing a couple small issues when you change to 2 or 4 columns.
image
image

**Comment by photodow** [comment]
- [ ] Percy is showing some really harsh lines. Any idea what's up here?

image

- [ ] The footer appears to be messed up again as well.

image

**Comment by IgnacioBecerra** [comment]
@photodow The harsh lines on percy seem to be the quick flash when rendering the component and calculating the lines I believe. Same with the footer, since it looks fine in the actual preview.

File: packages/web-components/src/components/card-group/card-group.ts [hunk 1]
```diff
@@ -201,63 +198,79 @@ class DDSCardGroup extends StableSelectorMixin(LitElement) {
...
           // last column
           if ((index + 1) % columns === 0) {
             e.style.paddingRight = '0';
-            const borderColor = getComputedStyle(document.body).getPropertyValue('--cds-ui-background');
-            e.style.borderRight = `inset 1px ${borderColor}`;
+            e.style.boxShadow = `0 0 0 ${borderColor}`;
           } else {
-            e.style.paddingRight = '1px';
-            e.style.borderRight = 'none';
+            // e.style.paddingRight = '1px';
```
Comment (emyarod) [inline packages/web-components/src/components/card-group/card-group.ts:247]:
```suggestion
```

File: packages/web-components/src/components/card-group/card-group.ts [hunk 2]
```diff
@@ -201,63 +198,79 @@ class DDSCardGroup extends StableSelectorMixin(LitElement) {
...
+        } else {
+          e.style.paddingLeft = '0';
+        }
+        // if one column and first item is empty then set top border for second item
+        if (columns === 1 && index === 1 && this._childItems[0].hasAttribute('empty')) {
+          e.style.paddingTop = `1px`;
+        }
+      } else {
+        e.removeAttribute('style');
+        // eslint-disable-next-line no-lonely-if
```
Comment (emyarod) [inline packages/web-components/src/components/card-group/card-group.ts:232]: I think we can refactor to avoid disabling this rule

**Comment by jeffchew** [comment]
Closing this, @kennylam will open a new PR with the changes.

**Pull request closed without merging**